File: 0283-move-zeroes/0283-move-zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
       x=0
       i=0
       n=len(nums)
       z=0
       while i<n:
        if nums[i]==0 :
             z+=1
        i+=1
       i=0
       while x<len(nums):
        if nums[x]!=0:
             nums[i]=nums[x]
             i+=1 
             x+=1
        else :
            x+=1
       j=n-z
       while j<n:
            nums[j]=0
            j+=1

File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
    int zcount=0;
    for(auto ele:nums) if(ele==0) zcount++;
    int i=0;
    int x=i;
    int n=nums.size();
    int j=n-zcount;
    while(x<n){
    if(nums[x]!=0){nums[i]=nums[x];x++;i++; }
    else x++;
    }
    while(j<n) nums[j++]=0;
    }
};